### PR TITLE
Buffs holy explosions

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

This PR buffs the anti-cultist and anti-revenant effects of holy explosions (which occur if you mix 75u+ of holy water and 75u+ of potassium together).

The cultist-stunning AoE effect of the holy water+potassium reaction now has a much longer range (7 tiles instead of around 4 tiles for a 100u/100u reaction) and occurs BEFORE the 2 second delay between the reaction of the holy water and the potassium and the actual explosion itself, making holy explosions actually effective against cultists (instead of being worse than a potassium+water explosion against them due to the delay between the reaction and the explosion). The message given to cultists who have been stunned and set on fire by a potassium+holy water reaction has been updated to reflect the fact that it now occurs before the "main" explosion.

Revenants now get stunned for an extra second by holy explosions, to ensure that they'll get caught in the "proper" blast.

## Why It's Good For The Game

Holy water explosions are actually worse than water+potassium ones against cultists, since the cultists can hear the loud sound made by the holy water+potassium reaction and use their 2 second grace period to easily saunter out of the explosion radius. The anti-cultist effect of holy explosions is rarely seen, since cultists who are dumb enough to remain within 4 or so tiles of a holy explosion will likely die from (or at least get critted by) the explosion itself anyway.

This PR cements holy explosions as an anti-cultist/revenant tool (the delay between the reaction and the explosion still makes them worse than potassium+water explosions against mobile non-cultists, which I think is perfectly fine), rather than as a cool but incredibly impractical weapon that only really has worth against objects (and sometimes revenants, if you can somehow time your grenade to trigger the reaction before the revenant just runs away).

## Changelog
🆑 ATHATH
balance: The holy water+potassium reaction's AOE cultist-stunning effect now affects a much larger area and occurs BEFORE the 2 second delay to the explosion.
tweak: The message given to cultists caught by this effect now reflects the fact that it now occurs before the "main" explosion.
balance: Revenants get stunned for an extra second by the revenant-stunning effect of holy water+potassium reactions.
/🆑

Cobby edit: used Dorsi's CL instead, yikes

### CODE REVIEW

**Comment by carlarctg** [comment]
please stop making your changelogs a wall of text

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 1]
```diff
@@ -59,23 +59,23 @@
 	if(created_volume >= 150)
 		playsound(get_turf(holder.my_atom), 'sound/effects/pray.ogg', 80, FALSE, round(created_volume/48))
 		strengthdiv = 8
+		var/deity
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: var/deity = GLOB.deity ? GLOB.deity : "Christ"

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 2]
```diff
@@ -59,23 +59,23 @@
...
-		sleep(20)
-		for(var/mob/living/carbon/C in get_hearers_in_view(round(created_volume/48,1),get_turf(holder.my_atom)))
+		for(var/mob/living/carbon/C in get_hearers_in_view(7,get_turf(holder.my_atom)))
 			if(iscultist(C))
-				to_chat(C, "<span class='userdanger'>The divine explosion sears you!</span>")
+				to_chat(C, "<span class='userdanger'>The power of [deity] compels you!</span>")
 				C.Paralyze(40)
 				C.adjust_fire_stacks(5)
 				C.IgniteMob()
+		sleep(20)
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: do not sleep, use timers.

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 3]
```diff
@@ -59,23 +59,23 @@
...
-			else
-				deity = "Christ"
 			to_chat(R, "<span class='userdanger'>The power of [deity] compels you!</span>")
-			R.stun(20)
+			R.stun(30)
 			R.reveal(100)
 			R.adjustHealth(50)
-		sleep(20)
-		for(var/mob/living/carbon/C in get_hearers_in_view(round(created_volume/48,1),get_turf(holder.my_atom)))
+		for(var/mob/living/carbon/C in get_hearers_in_view(7,get_turf(holder.my_atom)))
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: You should still make this volume based just less weak. for example created_volume / 25

**Comment by Dorsisdwarf** [comment]
🆑 ATHATH
balance: holy water+potassium reaction now AOE stuns cultists in a much longer range and occurs BEFORE the 2 second delay to the explosion.
tweak: The message given to cultists caught by this effect now reflects the fact that it now occurs before the "main" explosion.
balance: Revenants get stunned for an extra second by the revenant-stunning effect of holy water+potassium reactions.
/🆑

**Comment by AnturK** [comment]
:snowflake:

**Comment by ATH1909** [comment]
Thanks for reopening this, anturk

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: why are you sleeping for 20 seconds to do the base `on_reaction` to begin with?

Comment (ATH1909) [reply code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: Hey, I didn't write this code.

Also, this is the same thing that black powder does, so I figure that it's safe to keep this in (right?).

Comment (ATH1909) [reply code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: How would I make this use a timer, anyway?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: you need to figure out why this exists in the first place before you can figure out what timer to include

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 4]
```diff
@@ -59,23 +59,19 @@
...
-			var/deity
-			if(GLOB.deity)
-				deity = GLOB.deity
-			else
-				deity = "Christ"
+		var/deity = GLOB.deity ? GLOB.deity : "Christ"
+		for(var/mob/living/simple_animal/revenant/R in get_hearers_in_view(min(7,round(created_volume/25)),get_turf(holder.my_atom)))
 			to_chat(R, "<span class='userdanger'>The power of [deity] compels you!</span>")
-			R.stun(20)
+			R.stun(30)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]: stun bad

**Pull request closed without merging**